samples/batch/transcription-enabled-storage/CreateByTimerFunction/CreateTranscriptionByTimer.cs

```diff
@@ -0,0 +1,348 @@
+﻿// <copyright file="CreateTranscriptionByTimer.cs" company="Microsoft Corporation">
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE.md file in the project root for full license information.
+// </copyright>
+
+namespace CreateByTimerFunction
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Globalization;
+    using System.IO;
+    using System.Linq;
+    using System.Net;
+    using System.Text;
+    using System.Threading.Tasks;
+    using Connector;
+    using Microsoft.Azure.ServiceBus;
+    using Microsoft.Azure.ServiceBus.Core;
+    using Microsoft.Azure.WebJobs;
+    using Microsoft.Extensions.Logging;
+    using Newtonsoft.Json;
+
+    public static class CreateTranscriptionByTimer
+    {
+        private const int MaxMessagesPerRequest = 500;
+
+        private const int MaxFilesPerTranscriptionJob = 500;
+
+        private static DateTime StartDateTime;
+
+        private static ILogger Logger;
+
+        [FunctionName("CreateTranscriptionByTimer")]
+        public static async Task Run([TimerTrigger("0 */2 * * * *")] TimerInfo myTimer, ILogger log)
+        {
+            if (log == null)
+            {
+                throw new ArgumentNullException(nameof(log));
+            }
+
+            if (myTimer == null)
+            {
+                throw new ArgumentNullException(nameof(myTimer));
+            }
+
+            Logger = log;
+            StartDateTime = DateTime.UtcNow;
+            log.LogInformation($"C# Timer trigger function v3 executed at: {StartDateTime}. Next occurrence on {myTimer.Schedule.GetNextOccurrence(StartDateTime)}.");
+
+            if (!string.IsNullOrEmpty(CreateTranscriptionEnvironmentVariables.SecondaryLocale) && !CreateTranscriptionEnvironmentVariables.SecondaryLocale.Equals("None", StringComparison.OrdinalIgnoreCase))
+            {
+                log.LogWarning($"Language identification is not supported for timer-based transcription creation.");
+            }
+
+            var receiver = new MessageReceiver(CreateTranscriptionEnvironmentVariables.AzureServiceBus, "create_transcription_queue");
+
+            var totalServiceBusMessages = new List<Message>();
+
+            var serviceBusMessagesRemaining = true;
+            log.LogInformation("Pulling messages from queue...");
+            do
+            {
+                var messages = await receiver.ReceiveAsync(MaxMessagesPerRequest, TimeSpan.FromSeconds(20)).ConfigureAwait(false);
```

Comment:
here you basically pull in service messages for each run. What is the message expiration time? and how often the timer is running?

Reply:
So if many many messages arrive at the same time, it might stay in the queue for long time, we need proper expiration time to avoid that messages will be discarded due to expieration.

Reply:
There's a parameter for that when sending the service bus messages (MessageTTL). In theory, messages could stay forever in the queue - with basic tier, max. 14 days (https://docs.microsoft.com/de-de/azure/service-bus-messaging/message-expiration). I think even if a basic tier is selected, 14 days should be enough time to catch up with all messages.

What do you think?

Reply:
as discussed